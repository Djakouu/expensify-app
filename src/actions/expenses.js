import uuid from 'uuid'
import database from '../firebase/firebase'

export const addExpense = expense => ({
    type: 'ADD_EXPENSE',
    expense
});

export const startAddExpense = (expenseData = {}) => {
    return dispatch => {
        const {
            description = '',
            amount = 0,
            note = '', 
            createdAt = 0 
        } = expenseData;
        const expense = { description, amount, note, createdAt }
        return database.ref('expenses').push(expense).then((ref) => {
            dispatch(addExpense({
                id: ref.key,
                ...expense
            }));
        });
    };
};

// REMOVE EXPENSE
export const removeExpense = ({ id } = {}) => ({
    type: 'REMOVE_EXPENSE',
    id
})

export const startRemoveExpense =({ id } = {}) => {
    return dispatch => {
        return database.ref(`expenses/${id}`).remove().then(() => {
            dispatch(removeExpense({ id }));
        })
    }
};

// EDIT EXPENSE
export const editExpense = (id, updates) => ({
    type: 'EDIT_EXPENSE',
    updates,
    id
});

export const startEditExpense = (id, updates) => {
    return dispatch => {
        return database.ref(`expenses/${id}`).update(updates).then(() => {
            dispatch(editExpense(id, updates))
        });
    }
};

// SET EXPENSE
export const setExpenses = expenses => ({
    type: 'SET_EXPENSES',
    expenses
});

export const startSetExpenses = () => {
    return (dispatch => {
        return database.ref('expenses').once('value').then((snapshot) => {
            const expenses = [];
            snapshot.forEach(childSnapshot => {
                expenses.push({
                    id: childSnapshot.key,
                    ...childSnapshot.val()
                });
            });
            dispatch(setExpenses(expenses));
        });
    })
};