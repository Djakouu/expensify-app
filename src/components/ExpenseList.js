import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses'

export const ExpenseList = props => (
    <div>
        <h1>Expense List</h1>
        {props.expenses.length === 0 ? (
            <p> No expenses </p>
        ) : (
            props.expenses.map(expense => {
                return <ExpenseListItem key={expense.id} {...expense} />
            })
        )}
        
    </div>
);

const mapStatetoProps = state => ({
        expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStatetoProps)(ExpenseList);
