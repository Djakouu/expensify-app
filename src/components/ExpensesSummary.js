import React from 'react';
import { connect } from 'react-redux'
import numeral from "numeral";
import '../adjustments/numeral-locale-fr'
import getVisibleExpenses from '../selectors/expenses';
import { getExpensesTotal } from '../selectors/expenses-total';

export const ExpensesSummary = (props) => (
    <div>
        <h1>Viewing {props.expenses.length < 2 ? props.expenses.length + ' expense ' : props.expenses.length + ' expenses '} 
           totalling {numeral(getExpensesTotal(props.expenses) / 100).format('0,0.00 $')} 
        </h1>
    </div>
);

const mapStatetoProps = state => ({
    expenses: getVisibleExpenses(state.expenses, state.filters)
});

export default connect(mapStatetoProps)(ExpensesSummary);