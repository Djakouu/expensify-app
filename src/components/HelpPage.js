import React from 'react';

const Help = () => (
    <div>
        <p> This is from Help</p>
    </div>
);

export default Help;