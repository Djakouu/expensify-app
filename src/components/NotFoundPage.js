import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
    <div>
        <p>404 ­— <Link to="/">Go home</Link></p>
    </div>
);

export default NotFoundPage;