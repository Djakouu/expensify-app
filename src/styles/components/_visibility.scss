.show-for-mobile {
    @media (min-width: $desktop-breakpoint) {
        display: none;
    }
}

.show-for-desktop {
    @media (max-width: $desktop-breakpoint - .1rem) {
        display: none;
    }
}