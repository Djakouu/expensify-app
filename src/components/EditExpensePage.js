import React from 'react';
import ExpenseForm from './ExpenseForm'
import { connect } from 'react-redux'
import { editExpense, removeExpense } from '../actions/expenses'

export class EditExpense extends React.Component {
    onSubmit = expense => {
        this.props.editExpense(this.props.expense.id, expense);
        this.props.history.push('/')
    }
    onClick = () => {
        this.props.removeExpense({id: this.props.expense.id})
        this.props.history.push('/')
    }
    render() {
        return (
            <div>
                <h1>Edit Expense</h1>
                <ExpenseForm 
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                />
                <button onClick={this.onClick}>
                    Remove
                </button>
            </div>
        )
    }
}

// const Edit = (props) => ( 
//     <div>
//         <h1>Edit Expense</h1>
//         <ExpenseForm 
//             expense={props.expense}
//             onSubmit={(expense) => {
//                 props.dispatch(editExpense(props.expense.id, expense));
//                 props.history.push('/')
//             }}
//         />
//         <button onClick={() => {
//             props.dispatch(removeExpense({id: props.expense.id}));
//             props.history.push('/')
//         }
//         }>
//             Remove
//         </button>
//     </div>
// );

const mapStateToProps = (state, props) => ({
        expense: state.expenses.find(expense => expense.id === props.match.params.id)
});

const mapDispatchProps = (dispatch, props) => ({
        editExpense: (id, expense) => dispatch(editExpense(id, expense)),
        removeExpense: data => dispatch(removeExpense(data))
});

export default connect(mapStateToProps, mapDispatchProps)(EditExpense);