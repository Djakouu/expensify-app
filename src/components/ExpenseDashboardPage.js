import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const Expense = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default Expense;