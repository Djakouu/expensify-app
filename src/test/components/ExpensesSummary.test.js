import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render ExpensesSummary with one expense correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with two expenses correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[0], expenses[1]]} />);
    expect(wrapper).toMatchSnapshot();
});
