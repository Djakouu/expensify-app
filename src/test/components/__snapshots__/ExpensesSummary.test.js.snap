// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with one expense correctly 1`] = `
<div>
  <h1>
    Viewing 
    1 expense 
    totalling 
    1,00 €
  </h1>
</div>
`;

exports[`should render ExpensesSummary with two expenses correctly 1`] = `
<div>
  <h1>
    Viewing 
    2 expenses 
    totalling 
    3,00 €
  </h1>
</div>
`;
